docs/Implementing-easyblocks.rst

```diff
@@ -0,0 +1,486 @@
+sanity check is also included.
+
+Since EasyBuild v2.0, using a generic easyblock requires specifying the ``easyblock`` parameter in the easyconfig file;
+if it is not specified, EasyBuild will try and find the software-specific easyblock derived from the software name.
+
+The distinction between generic and software-specific easyblocks can be made based on the naming scheme that is used
+for an easyblock, see also :ref:`implementing_easyblocks_naming_scheme`.
+
+
+.. _implementing_easyblocks_vs_easyconfigs:
+
+Easyblocks vs easyconfigs
+-------------------------
+
+Before you start implementing an easyblock, you should determine whether or not you really need an easyblock.
+
+:ref:`easyconfig_files` provide quite a bit of flexibility that alleviate the need for implementing many (software-specific)
+easyblocks. Examples include easyconfig parameters like ``(pre){config,build,installopts}`` (available for any easyblock),
+``install_cmd`` (only for ``Binary`` generic easyblock or derivates), etc. See :ref:`writing_easyconfig_files` for an
+overview of what is supported in easyconfig files, and :ref:`generic_easyblocks` for custom easyconfig parameters for
+the existing generic easyblocks.
+
+On the other hand, somewhat complex or heavily customised software build and installation procedures may be handled better
+via a custom easyblock.
+
+Easyblocks are "do once and forget", and can provide a *central* solution for peculiarities of installations.
+Since they are basically Python scripts, they are very flexible and can take care of the
+'heavy lifting' that goes beyond the key-value defining scope of easyconfig files.
+
+Hence, there is a fine line between using 'fat' easyconfigs with a generic easyblock and using a custom software-specific easyblock.
+
+Reasons to consider implementing a custom easyblock include:
+
+* 'critical' values for easyconfig parameters, which are required to make the installation succeed
+* compiler- or toolchain-specific aspects of the build and installation procedure (e.g., configure/build/install options)
+* interactive commands that need to be run
+* custom (configure) options for dependencies
+* having to create or adjust specific (configuration) files
+* 'hackish' usage of existing (generic) easyblocks and available easyconfig parameters
+
+One important aspect to consider of course is that implementing easyblocks requires some familiarity with Python,
+while easyconfig files can mostly be seen as a set of pure key-value definitions and hence are easier to create or
+update.
+
+
+.. _implementing_easyblocks_naming_scheme:
+
+Naming scheme for easyblocks
+----------------------------
+
+Easyblocks need to follow a strict naming scheme, to ensure that EasyBuild can pick them up as needed.
+This involves two aspects: i) the name and location of the Python module file,
+and ii) the name of the Python class in that module.
+
```

Comment:
I find it weird that you mention first the name and location and then the class, but the following subsections are in the reversed order. Also, I think it makes sense to talk about class after talking about location. So, reorder the subsections?